src/sqlfluff/core/parser/segments/indent.py

```diff
@@ -105,4 +118,33 @@ class Dedent(Indent):
 
     """
 
+    type = "dedent"
     indent_val = -1
+
+
+class TemplateSegment(MetaSegment):
+    """A segment which is empty but indicates something should be.

```

Comment:
Just wanted to be clear on the docstring here. How come the `source_str` is optional? Is there ever cases where we don't store the original string?

Reply:
Goooood question. I'm not sure there is. Maybe I should revisit that. 🤔

Reply:
This now raises an exception if not set. I think we'll rely on it later for some of the rules we want to do so best to make sure we can rely on it.